packages/flutter_tools/test/asset_bundle_package_test.dart

```diff
@@ -133,6 +133,32 @@ $assetsSection
     }, overrides: <Type, Generator>{
       FileSystem: () => new MemoryFileSystem(),
     });
+    
+    testUsingContext('One package with one asset not specified', () async {
+      establishFlutterRoot();
+
+      final List<String> assetEntries = <String>['packages/test_package/a/foo'];
+      writePubspecFile(
+        'pubspec.yaml',
+        'test',
+        assets: assetEntries,
+      );
+      writePackagesFile('test_package:p/p/lib/');
+      writePubspecFile('p/p/pubspec.yaml', 'test_package');
+
+      final List<String> assets = <String>['a/foo'];
+      writeAssets('p/p/lib/', assets);
+
+      final String expectedAssetManifest = '{"packages/test_package/a/foo":'
+          '["packages/test_package/a/foo"]}';
+      await buildAndVerifyAssets(
+        assets,
+        <String>['test_package'],
+        expectedAssetManifest,
+      );
+    }, overrides: <Type, Generator>{
+      FileSystem: () => new MemoryFileSystem(),
+    });
```

Comment:
[nit] You could extract the context overrides map as getter in the ambient `group`:
```
  Map<Type, Generator> get contextOverrides => <Type, Generator>{
    FileSystem: () => new MemoryFileSystem()
  };
```
and reuse it across calls to `testUsingContext`:
```
  testUsingContext('some test', () {
     ...
  }, overrides: contextOverrides);
```